railties/lib/rails/application/configuration.rb

```diff
@@ -287,6 +287,7 @@ def load_defaults(target_version)
             active_record.raise_on_assign_to_attr_readonly = true
             active_record.belongs_to_required_validates_foreign_key = false
             active_record.before_committed_on_all_records = true
+            active_record.default_column_serializer = nil
```

Comment:
This should be in 7.0, and setting it to YAML. Whether we want to be broken out of the box for new applications is debatable, but we definitely don't want to break for upgrades.

Reply:
@matthewd the default is set here: https://github.com/rails/rails/blob/50edf85acd9ab452c5d8ec6913e8829e7f36cc16/activerecord/lib/active_record/attribute_methods/serialization.rb#L19

It's still YAML for all existing applications, we only set it to nil if you have `load_default 7.1`.

Reply:
This causes different behaviour of active_record when used by itself vs being used in a Rails 7.1 application. Would it not be better to have active_record default to `nil` also so at least the behaviour is consistent and gems which depend on active_record can have parity between their tests and behaviour when used in Rails apps (see https://github.com/rpush/rpush/pull/668 for example)?

Reply:
@benlangfeld unfortunately no, because the default must be YAMLColumn if `load_defaults` is never called. 

This is the way we do it for all default changes.